DESIGN.md

```diff
@@ -0,0 +1,73 @@
+# Extensions
+
+*Note*: This document is evolving and is in draft state.
+
+Plugin architecture enables extending core features of OpenSearch. There are various kinds of plugins which are supported.
+But, the architecture has significant problems for OpenSearch customers. Importantly, plugins can fatally impact the cluster
+i.e critical workloads like ingestion/search traffic would be impacted because of a non-critical plugin like s3-repository failed with an exception.
+
+This problem is exponentially grows when we would like to run a 3rd Party plugin from the community.  
+As OpenSearch and plugins run in the same process, it brings in security risk, dependency conflicts and reduces the velocity of releases.
+
+Introducing extensions, a simple and easy way to extend features of OpenSearch. It would support all plugin features and enable them to run in a different process or another node via OpenSearch SDK.
+
+Meta Issue: [Steps to make OpenSearch extensible](https://github.com/opensearch-project/OpenSearch/issues/2447)  
+Sandboxing: [Step towards modular architecture in OpenSearch](https://github.com/opensearch-project/OpenSearch/issues/1422)  
+Security: [Security for extensions](SECURITY.md) 
+
+## Plugins Architecture
+
+![](Docs/plugins.png)
+
+Plugins are installed via [`opensearch-plugin`](https://github.com/opensearch-project/OpenSearch/blob/main/distribution/tools/plugin-cli/src/main/java/org/opensearch/plugins/InstallPluginCommand.java) and are class loaded into OpenSearch.
```

Comment:
```suggestion
Plugins are installed via [`opensearch-plugin`](https://github.com/opensearch-project/OpenSearch/blob/main/distribution/tools/plugin-cli/src/main/java/org/opensearch/plugins/InstallPluginCommand.java) plugin and are class loaded into OpenSearch.
```
We can link any plugin's build.gradle file here for more visibility having opensearch-plugin

Reply:
This is not the gradle plugin. This is `opensearch-plugin` binary tool which is already linked here. 
Are you looking for something else?